sxg_rs/src/lib.rs

```diff
@@ -247,9 +249,46 @@ impl SxgWorker {
         let headers = Headers::new(fields, &self.config.strip_request_headers);
         headers.forward_to_origin_server(accept_filter, &self.config.forward_request_headers)
     }
-    pub fn validate_payload_headers(&self, fields: HeaderFields) -> Result<()> {
+    /// Checks `fields` as response headers from backend server,
+    /// and returns the reqsponse headers to be sent to browser.
+    pub fn transform_payload_headers(&self, fields: HeaderFields) -> Result<Headers> {
         let headers = Headers::new(fields, &self.config.strip_response_headers);
-        headers.validate_as_sxg_payload()
+        headers.validate_as_sxg_payload()?;
+        Ok(headers)
+    }
+    #[cfg(feature = "rust_signer")]
+    pub fn create_rust_signer(&self) -> Result<signature::rust_signer::RustSigner> {
+        let private_key_der = base64::decode(
+            self.config
+                .private_key_base64
+                .as_ref()
+                .ok_or(Error::msg("Config private_key_base64 is not set"))?,
+        )?;
+        const KEY_SIZE: usize = 32;
+        if private_key_der.len() != KEY_SIZE {
+            return Err(Error::msg(format!(
+                "Expecting byte size of private_key_base64 to be {}, found {}",
+                KEY_SIZE,
+                private_key_der.len()
+            )));
+        }
+        Ok(signature::rust_signer::RustSigner::new(&private_key_der))
```

Comment:
Maybe while you're here, get rid of the `unwrap` in `RustSigner::new` and change it to return a `Result<Self>`.

Reply:
Done.